lib/retrieve_package.js

```diff
@@ -56,18 +55,33 @@ function place_binary(options, callback) {
   }).on('close', afterTarball).on('error', callback));
 }
 
-function checkLibraryExists(options, callback) {
-  existsAsync(options.dst, function(found) {
-    if (found && !options.update_binary) {
+function checkLibraryExists(options, callback) {  
+  try{
+    var found = fse.pathExistsSync(options.dst);
+  } catch(e){
+    return callback(e);
+  }
+
+  if (found) {
+  // this handles a situation where a download fails and 
```

Comment:
Can we delete the directory immediately after a download fails so other methods don't have to have checks like this?

Reply:
This is handling the case when someone uses `Ctrl + C` to exit the process - I don't think it would never hit an error handler. I'm not sure how you would "listen" for that cancellation and remove empty directories?